apps/service_providers/migration_utils.py

```diff
@@ -0,0 +1,44 @@
+from apps.service_providers.llm_service.default_models import LLM_MODEL_PARAMETERS
+
+
+def populate_temperature_params(Node, LlmProviderModel):
+    """
+    This migration prepares the addition of the temperature parameter to the LLM model parameters.
+
+    All nodes that uses LLM models that supports temperature should be updated so that its `llm_model_parameters`
+    parameter contains the same value as the existing toplevel `llm_temperature` value.
+
+    For now we're not going to remove the existing `llm_temperature` toplevel parameter, just to be safe.
+    It should be OK to leave it there indefinitely, since we wouldn't be using it anymore.
+    """
+
+    models_supporting_temperature = []
+    for model, param_cls in LLM_MODEL_PARAMETERS.items():
+        if "temperature" in param_cls.model_fields:
+            models_supporting_temperature.append(model)
+
+    supported_model_ids = list(
+        LlmProviderModel.objects.filter(name__in=models_supporting_temperature).values_list("id", flat=True)
+    )
+
+    nodes_to_save = []
+
+    # Some ids are saved as strings, others as ints, so we need to check for both
```

Comment:
I'm going to investigate why this is. There are ~5k+ integer ones and ~3k+ string ones if I remember correctly. It's not technically an issue, but it makes data migrations (and reading ids) from `node.params` a pain, since we have to account for both cases